tools/migration/mapprojects

```diff
@@ -0,0 +1,211 @@
+        usage = \
+            'usage: %prog [options] <dbuser> <dbpwd> <mapprojectsfile>'
+        parser = OptionParser(usage)
+        parser.add_option('-u', '--dbuser', dest='dbuser',
+                          help='db user')
+        parser.add_option('-p', '--dbpwd', dest='dbpwd',
+                          help='db password')
+        parser.add_option('-m', '--mapprojectsfile',
+                          dest='mapprojectsfile',
+                          help='the path of mapping projects file')
+
+        (options, args) = parser.parse_args()
+        return (options.dbuser, options.dbpwd, options.mapprojectsfile)
+
+    def init_from_input(self):
+        (self.dbuser, self.dbpwd, self.mapprojectsfile) = \
+            Parameters.parse_input()
+
+
+class AccessLog:
+
+    def __init__(self, project_id, log_id):
+        self.project_id = project_id
+        self.log_id = ''
+
+
+class Repository:
+
+    def __init__(self, project_id, repository_id):
+        self.project_id = project_id
+        self.repository_id = ''
+
+
+class Project:
+
+    def __init__(
+        self,
+        project_id,
+        name,
+        index_id,
+        ):
+        self.project_id = project_id
+        self.project_name = name
+        self.index_id = index_id
+
+
+class HarborUtil:
+
+    def __init__(self, dbuser, dbpwd):
+        self.serverName = 'localhost'
+        self.user = dbuser
+        self.password = dbpwd
+        self.port = '3306'
+        self.subDB = 'registry'
+        self.db = None
+        self.cursor = None
+
+    def connect(self):
+        try:
+            self.db = MySQLdb.connect(host=self.serverName,
+                    user=self.user, passwd=self.password, db=self.subDB)
+            self.db.autocommit(False)
+            self.cursor = self.db.cursor()
+        except Exception, e:
+            raise Exception(e)
+
+    def close(self):
+        try:
+            self.db.commit()
+            self.cursor.close()
+            self.db.close()
+        except Exception, e:
+            print str(e)
+
+    def enable_foreign_key_check(self):
+        try:
+            self.cursor.execute('SET FOREIGN_KEY_CHECKS=1')
+        except Exception, e:
+            print str(e)
+
+    def disable_foreign_key_check(self):
+        try:
+            self.cursor.execute('SET FOREIGN_KEY_CHECKS=0')
+        except Exception, e:
+            print str(e)
+
+    def update_access_log_table(self, projects):
+        for project in projects:
+            access_logs = []
+            try:
+                query_log_id = \
+                    'SELECT * from registry.access_log where project_id=%s' \
+                    % project.project_id
+                self.cursor.execute(query_log_id)
+                self.cursor.fetchall()
+                for result in self.cursor:
+                    access_logs.append(result[0])
+            except Exception, e:
+                raise Exception(e)
+
+            for item in access_logs:
```

Comment:
nested loop will not handle collision correctly.

Same comments to update_repository_table and  update_replication_policy_table